### PR TITLE
refactor: Refactor seats logic

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

- Refactors `handleSeats` logic into separate files and into a new folder structure
image
- Adds unit tests for these logic blocks, replacing some E2E tests. Testing goes from 6mins down to 19 seconds

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] Tests (Unit/Integration/E2E or any other test)

## How should this be tested?

- Run booking seats E2E and unit tests

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/bookings/lib/handleSeats/handleSeats.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  } else {
+    resultBooking = await createNewSeat(newSeatedBookingObject, seatedBooking);
+  }
+
+  // Here we should handle every after action that needs to be done after booking creation
+
+  // Obtain event metadata that includes videoCallUrl
+  const metadata = evt.videoCallData?.url ? { videoCallUrl: evt.videoCallData.url } : undefined;
+  try {
+    await scheduleWorkflowReminders({
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleSeats/handleSeats.ts:94]: same as for `handleWebhookTrigger` below, workflow is triggered twice on reschedule when booking seat

**Review by CarinaWolli — DISMISSED** [review]

File: packages/features/bookings/lib/handleSeats/reschedule/attendee/attendeeRescheduleSeatedBooking.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import type EventManager from "@calcom/core/EventManager";
+import { sendRescheduledSeatEmail } from "@calcom/emails";
+import prisma from "@calcom/prisma";
+import type { Person, CalendarEvent } from "@calcom/types/Calendar";
+
+import { findBookingQuery } from "../../../handleNewBooking";
+import lastAttendeeDeleteBooking from "../../lib/lastAttendeeDeleteBooking";
+import type { RescheduleSeatedBookingObject, SeatAttendee, NewTimeSlotBooking } from "../../types";
+
+const attendeeRescheduleSeatedBooking = async (
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleSeats/reschedule/attendee/attendeeRescheduleSeatedBooking.ts:13]: Why is that not used in `rescheduleSeatedBooking`?

File: packages/features/bookings/lib/handleSeats/create/createNewSeat.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        isHostConfirmationEmailsDisabled = allowDisablingHostConfirmationEmails(workflows);
+      }
+
+      if (isAttendeeConfirmationEmailDisabled) {
+        isAttendeeConfirmationEmailDisabled = allowDisablingAttendeeConfirmationEmails(workflows);
+      }
+    }
+    await sendScheduledSeatsEmails(
+      copyEvent,
+      invitee[0],
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleSeats/create/createNewSeat.ts:136]:
```suggestion
      inviteeToAdd,
```

File: packages/lib/errorCodes.ts [hunk 4]
```diff
@@ -6,4 +6,10 @@ export enum ErrorCode {
   AlreadySignedUpForBooking = "already_signed_up_for_this_booking_error",
   HostsUnavailableForBooking = "hosts_unavailable_for_booking",
   EventTypeNotFound = "event_type_not_found_error",
+  BookingNotFound = "booking_not_found_error",
+  BookingSeatsFull = "booking_seats_full_error",
+  MissingPaymentCredential = "missing_payment_credential_error",
+  MissingPaymentAppId = "missing_payment_app_id_error",
+  UpdateDatedBookingNotFound = "update_dated_booking_not_found_error",
```
Comment (CarinaWolli) [inline packages/lib/errorCodes.ts:13]: I think this translation key is missing in common.json

Comment (joeauyeung) [reply packages/lib/errorCodes.ts]: This error code wasn't being used anywhere so removed for now.

File: packages/features/bookings/lib/handleSeats/types.d.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import type { AppsStatus } from "@calcom/types/Calendar";
+
+import type { Booking } from "../handleNewBooking";
+
+export type BookingSeat = Prisma.BookingSeatGetPayload<{ include: { booking: true; attendee: true } }> | null;
+
+export type NewSeatedBookingObject = {
+  rescheduleUid: string | undefined;
+  reqBookingUid: string | undefined;
+  eventType: NewBookingEventType;
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleSeats/types.d.ts:12]: Some imports are missing in the files which causes type errors

for example, import for `NewBookingEventType` and `OriginalRescheduledBooking`

File: packages/features/bookings/lib/handleSeats/reschedule/rescheduleSeatedBooking.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
+import type {
+  HandleSeatsResultBooking,
+  SeatedBooking,
+  RescheduleSeatedBookingObject,
+  SeatAttendee,
+} from "bookings/lib/handleSeats/types";
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleSeats/reschedule/rescheduleSeatedBooking.ts:6]:
```suggestion
} from "../types";
```

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
After type errors are fixed, I can approve 👍

EDIT: There are failing unit and e2e tests as well which might be related to the changes (https://github.com/calcom/cal.com/pull/12905/files#r1443076439)

File: packages/features/bookings/lib/handleSeats/types.d.ts (hunk 5 above)
Comment (joeauyeung) [reply packages/features/bookings/lib/handleSeats/types.d.ts:12]: Thank you! I was trying to figure out why it wasn't working. I don't get why my IDE wasn't throwing an error though.

**Review by CarinaWolli — APPROVED** [review]
Great work @joeauyeung 👏🏻

**Pull request merged**